grpc/src/main/java/com/linecorp/armeria/server/grpc/GrpcServiceBuilder.java

```diff
@@ -815,16 +815,18 @@ public GrpcServiceBuilder enableHealthCheckService(boolean enableHealthCheckServ
     }
 
     /**
-     * Sets the specified {@link GrpcStatusFunction} that maps a {@link Throwable} to a gRPC {@link Status}.
+     * Sets the specified {@link GrpcExceptionHandlerFunction} that maps a {@link Throwable}
+     * to a gRPC {@link Status}.
      *
      * <p>Note that this method and {@link #addExceptionMapping(Class, Status)} are mutually exclusive.
+     *
      */
-    public GrpcServiceBuilder exceptionMapping(GrpcStatusFunction statusFunction) {
-        requireNonNull(statusFunction, "statusFunction");
+    public GrpcServiceBuilder exceptionMapping(GrpcExceptionHandlerFunction grpcExceptionHandlerFunction) {
```

Comment:
This will be a breaking change that we shouldn't make.
I think we have two options to solve it:
- Reverting this method and making `GrpcExceptionHandlerFunction` extend `GrpcStatusFunction` as we do for `StreamDecoderFactory`
  - `public interface GrpcExceptionHandlerFunction extends GrpcStatusFunction {...}`
- Add both methods that take `GrpcExceptionHandlerFunction` and `GrpcStatusFunction` and call another `exceptionMapping` by converting `GrpcStatusFunction` to `GrpcExceptionHandlerFunction`:
  - ```java
     @Deprecated
     public GrpcServiceBuilder exceptionMapping(GrpcStatusFunction grpcStatusFunction) {
         return exceptionMapping(toHandlerFunction(grpcStatusFunction));
     }

     public GrpcServiceBuilder exceptionMapping(GrpcExceptionHandlerFunction grpcExceptionHandlerFunction) {...}
I slightly prefer the second way because we can just remove the deprecated method when we update the major version. Please, let me know your opinion. 😄

Reply:
I agree with your opinion as well.
I tried the second option but encountered a problem.
As `GrpcExceptionHandlerFunction` and `GrpcStatusFunction` have the same function definition,
If a user defines this builder argument as a lambda expression, A compiler cannot distinguish them.

If we change `GrpcExceptionHandlerFunction` to the following you suggested, the problem might be resolved.
`public interface GrpcExceptionHandlerFunction extends GrpcStatusFunction {...}`

So let me try it.

Reply:
Ah, I didn't notice that. We could probably have different method names for `GrpcExceptionHandlerFunction` but not sure if it's a good idea. So, Let's try with the first option. 😉

Reply:
I tried the first option but there is a compile error in Scala code.
It seems that Scala can't distinguish these methods.
So, I think we need to make this breaking change.
WDYT?

Reply:
Oops, then let's just use a different method name as @ikhoon suggested. 😉 
https://github.com/line/armeria/pull/5046#discussion_r1332733707